### PR TITLE
Fix --emit llvm-ir when asked to emit binary mode

### LINKED ISSUE #4054: [BUG]: "LLVM IR" view fails for rust with "Compile to binary" on
### Describe the bug

The LLVM IR view isn't working for Rust, but this doesn't look exactly the same as https://github.com/compiler-explorer/compiler-explorer/issues/2165 and it seems to happen for all versions of rust that I try.

### Steps to reproduce

1. Visit godbolt.org
2. Select Rust and some version of `rustc`
3. Put something with a `main` into the input box, e.g.:

    ```rust
    // Type your code here, or load an example.
    pub fn square(num: i32) -> i32 {
    num * num
    }
    
    // If you use `main()`, declare it as `pub` to see it in the output:
    pub fn main() {
    }
    ```

4. Turn on "Output > Compile to binary"
6. Click "Add New > LLVM IR"

### Expected behavior

The LLVM IR should be there, or at least a reasonable explanation of why not.

### Reproduction link

https://godbolt.org/z/c7xr4Pjq5

### Screenshots

As you can see, it just prints this instead:
```
Internal error; unable to open output path
```
image

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by RubenRBS**
This _might_ be the fix for the linked issue, as I wrote in the original issue:

The issue is that we expected rustc to always output the llvm-ir in a `output.ll`, as we normally call it with `-o output.s --emit asm --emit llmv-ir`, which makes it warn us with:
```
warning: due to multiple output types requested, the explicitly specified output file name will be adapted for each output type

warning: 1 warning emitted
```

and then rustc goes ahead and creates both the `output.s` and `output.ll` files.

In the binary case, we only ask it to emit llvm-ir, so it interprets that the passed `-o output.s` is for the llvm-ir. In hindsight it makes some sense, but this flew under our radar until now

Now the issue is that, afaik (Im no expert in llvm), we now get the expected ir:
imagen

but most of the file is binary nonsense:
imagen

Closes #4054

### CODE REVIEW

File: lib/base-compiler.ts [hunk 1]
```diff
@@ -1011,12 +1011,13 @@ export class BaseCompiler {
         if (output.code !== 0) {
             return [{text: 'Failed to run compiler to get IR code'}];
         }
+        console.log(newOptions, output);
```
Comment (partouf) [inline lib/base-compiler.ts:1014]: *cough*

Comment (RubenRBS) [reply lib/base-compiler.ts:1014]: Ey you found a bug! Every check passed both locally and in the CI. I'll fix it in another PR, thanks for checking the code :)

**Comment by partouf** [comment]
The binary bits don't look like something I've seen before, but maybe the llvm tools can still split them up somehow. I'll have a look

**Comment by partouf** [comment]
binary and llvmir don't really seem to mix at all, but this is better than nothing

**Comment by partouf** [comment]
this is live

**Pull request merged**